### DISCUSSION TITLE
How to use positions in the mesh inside a function

### DISCUSSION THREAD

**Discussion by MScKarl** [idaholab/moose#20848] [Q&A General] [answered]
Hi!
I am using coord_type = RZ, for a polar mesh.

I want to use an equation that is dependent on space coordinates r and θ in my mesh. Is there an easy way to reference the positions in a function, I am thinking ParsedMaterial?
If I need to use the absolute value and the sign function (separately), do I write them the same in moose as I would in a C++ code? Is there a handbook where I can search this thing, I couldn’t find anything on Google or the website?

Best regards
Karl

**Comment by GiudGiud** [comment, marked as answer]
Hello

So RZ is RZ not Rtheta, we dont actually support Rtheta right now. Closest thing is actually 2D-XY really. 

For positions, you can pass x, y, z in parsed expressions. In the parsed material it might not process it right but you can define a function, like a parsed function
https://mooseframework.inl.gov/source/functions/MooseParsedFunction.html
and then pass it in the parsed material. 

The fparser documentation (the parser we use) is there: https://fparser.readthedocs.io/en/latest/

Absolute should be |x| I think. and sign might not be supported, if you can write : if(x<0,-1,1)

Guillaume

**Reply by GiudGiud** [reply]
also note in RZ, x is R and y is Z, so they should be referred as such in the parsed expressions
